etna/datasets/tsdataset.py

```diff
@@ -1210,33 +1213,14 @@ def train_test_split(
         if train_start_defined < self.df.index.min():
             warnings.warn(f"Min timestamp in df is {self.df.index.min()}.")
 
-        train_df = self.df.loc[train_start_defined:train_end_defined][self.raw_df.columns]  # type: ignore
-        train_raw_df = self.raw_df.loc[train_start_defined:train_end_defined]  # type: ignore
-        train = TSDataset(
-            df=train_df,
-            df_exog=self.df_exog,
-            freq=self.freq,
-            known_future=self.known_future,
-            hierarchical_structure=self.hierarchical_structure,
-        )
-        train.raw_df = train_raw_df
-        train._regressors = deepcopy(self.regressors)
-        train._target_components_names = deepcopy(self.target_components_names)
-        train._prediction_intervals_names = deepcopy(self._prediction_intervals_names)
-
-        test_df = self.df.loc[test_start_defined:test_end_defined][self.raw_df.columns]  # type: ignore
-        test_raw_df = self.raw_df.loc[train_start_defined:test_end_defined]  # type: ignore
-        test = TSDataset(
-            df=test_df,
-            df_exog=self.df_exog,
-            freq=self.freq,
-            known_future=self.known_future,
-            hierarchical_structure=self.hierarchical_structure,
-        )
-        test.raw_df = test_raw_df
-        test._regressors = deepcopy(self.regressors)
-        test._target_components_names = deepcopy(self.target_components_names)
-        test._prediction_intervals_names = deepcopy(self._prediction_intervals_names)
+        train = deepcopy(self)
```

Comment:
I think we should come up with a more economical solution in terms of resource usage. Currently here we might occupy up to 3x of memory compared to the original size. Also here we are unnecessarily copying `df_exog`  when reference to the original could be used. Here we could try to utilize pandas copy/view semantics instead of explicitly copying dataframes.

Reply:
I added new version of code, but it should be discussed: there are some possible problems.